File: echomind_app/lib/models/question.dart
import 'package:json_annotation/json_annotation.dart';

part 'question.g.dart';

@JsonSerializable()
class Question {
  final String id;
  @JsonKey(name: 'image_url')
  final String? imageUrl;
  @JsonKey(name: 'is_correct')
  final bool? isCorrect;
  final String source;
  @JsonKey(name: 'diagnosis_status')
  final String diagnosisStatus;
  @JsonKey(name: 'created_at')
  final String? createdAt;

  const Question({
    required this.id,
    this.imageUrl,
    this.isCorrect,
    required this.source,
    required this.diagnosisStatus,
    this.createdAt,
  });

  factory Question.fromJson(Map<String, dynamic> json) =>
      _$QuestionFromJson(json);
}

@JsonSerializable()
class HistoryDateGroup {
  final String date;
  final List<Question> questions;

  const HistoryDateGroup({required this.date, required this.questions});

  factory HistoryDateGroup.fromJson(Map<String, dynamic> json) =>
      _$HistoryDateGroupFromJson(json);
}

File: backend/app/schemas/question.py
"""Question schemas."""
from datetime import datetime
from pydantic import BaseModel


class QuestionUploadRequest(BaseModel):
    image_url: str
    is_correct: bool | None = None
    source: str = "manual"


class QuestionResponse(BaseModel):
    id: str
    image_url: str | None
    is_correct: bool | None
    source: str
    diagnosis_status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class HistoryDateGroup(BaseModel):
    date: str
    questions: list[QuestionResponse]
